### PR TITLE
[ISSUE #3713]Add new plugin: key-auth

### PR DESCRIPTION
**Pull request by Albertsirius**
#3713 Add new plugin: key-auth

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by yu199195** [comment]
pls check ci

**Comment by Albertsirius** [comment]
> pls check ci
屏幕截图 2022-07-26 151141
屏幕截图 2022-07-26 151238

I don't know why it failed. I didn't modify this source.

**Comment by Albertsirius** [comment]
Hi, @yu199195, it hasn't been reviewed, any problems?

**Comment by Albertsirius** [comment]
@yu199195 Have modified and fixed the SQL scripts. Pls reviewed.

File: shenyu-plugin/shenyu-plugin-key-auth/src/main/java/org/apache/shenyu/plugin/key/auth/KeyAuthPlugin.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * @return whether the key is correct.
+     */
+    private boolean checkKey(final ServerWebExchange exchange, final String keyName, final String key) {
+        HttpHeaders httpHeaders = exchange.getRequest().getHeaders();
+        if (StringUtils.equals(httpHeaders.getFirst(keyName), key)) {
+            return true;
+        }
+        MultiValueMap<String, String> multiValueMap = exchange.getRequest().getQueryParams();
+        if (StringUtils.equals(multiValueMap.getFirst(keyName), key)) {
+            return true;
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-key-auth/src/main/java/org/apache/shenyu/plugin/key/auth/KeyAuthPlugin.java:83]: StringUtils.equals(httpHeaders.getFirst(keyName), key) 

&&  StringUtils.equals(multiValueMap.getFirst(keyName), key)

return true？

File: db/init/mysql/schema.sql [hunk 2]
```diff
@@ -599,6 +599,7 @@ INSERT INTO `plugin` VALUES ('34', 'loggingAliyunSls','{\"projectName\": \"sheny
 INSERT INTO `plugin` VALUES ('35', 'loggingPulsar', '{\"topic":\"shenyu-access-logging\", \"serviceUrl\": \"pulsar://localhost:6650\"}', 'Logging', 185, 0, '2022-06-30 21:00:00', '2022-06-30 21:00:00');
 INSERT INTO `plugin` VALUES ('36', 'loggingTencentCls','{\"endpoint\": \"ap-guangzhou.cls.tencentcs.com\", \"topic\": \"shenyu-topic\"}', 'Logging', 176, 0, '2022-06-30 21:00:00', '2022-06-30 21:00:00');
 INSERT INTO `plugin` VALUES ('38', 'loggingClickHouse', '{\"host\":\"127.0.0.1\",\"port\":\"8123\",\"databse\":\"shenyu-gateway\",\"username\":\"foo\",\"password\":\"bar\"}', 'Logging', 195, 0, '2022-06-30 21:00:00', '2022-06-30 21:00:00');
+INSERT INTO `plugin` VALUES ('39', 'keyAuth', NULL, 'Authentication', 430, 0, '2022-07-24 19:00:00', '2022-07-24 19:00:00');
```
Comment (yu199195) [inline db/init/mysql/schema.sql:602]: maybe miss ruleHandler？

Comment (yu199195) [reply db/init/mysql/schema.sql:602]: sort is not 430, Should be small.

Comment (Albertsirius) [reply db/init/mysql/schema.sql:602]: OK, I will choose a small sort number.

**Comment by Albertsirius** [comment]
@yu199195 Fixed the conflicting files

**Review by yu199195 — APPROVED** [review]

**Pull request merged**